tests/components/otbr/test_websocket_api.py

```diff
@@ -446,6 +578,54 @@ async def test_set_network_fails_3(
     assert msg["error"]["code"] == "set_enabled_failed"
 
 
+async def test_set_network_fails_4(
+    hass: HomeAssistant,
+    aioclient_mock: AiohttpClientMocker,
+    otbr_config_entry_multipan,
+    websocket_client,
+) -> None:
+    """Test create network."""
+    with patch(
+        "python_otbr_api.OTBR.get_border_agent_id",
+        side_effect=python_otbr_api.OTBRError,
```

Comment:
I'd make a pytest fixture for this patch instead since we always need to either set a return value (the default) or a side effect.

Reply:
This would indeed be a good improvement, let's do that in a follow-up PR though since all the existing tests would benefit from this and should be rewritten accordingly 👍